lib/instrumentation/http.js

```diff
@@ -20,47 +20,85 @@ let instrumentHTTP = (http, opts = {}) => {
     // node 10+ supports a three arg form.  Instead of writing both implementations,
     // dynamically switch on arg types.  I _hate_ code that does this, but it's easier
     // and cleaner than the alternative.
-    return function (_url, options, cb) {
+    //
+    // The http.request function can be called either as:
+    // http.request(options[, callback])
+    // or
+    // http.request(url[, options][, callback])
+    // So we can have 1-3 args to work with. For more details, see:
+    // http://nodejs.org/api/http.html#http_http_request_url_options_callback
+    return function (/* Args are parsed using the local built-in variable "arguments" */) {
```

Comment:
Looks like HC targets node 10+, we could use [Rest Parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters#browser_compatibility) here. Not sure if they have a preference around that (I do not), but `arguments` is arguably a bit of a wart in JS, because it's an "array-like object" and that can bite you or add boilerplate like the `Array.from(arguments)` when you need to to actually be an array.

Reply:
yeah. this isn't a bad idea! I appreciate the subtlety with the built in vs. a variadic set of args.

Reply:
Oh yes, you can still have named parameters when using `arguments.length` to check arity. I might lean that way to match the naming in the node docs.

I'd also name this anon function (`wrappedRequest` or similar) because it makes debugging easier, but they are clearly not fussed about that.